src/views/InvolvementProfile/components/Membership/components/AdminCard/index.js

```diff
@@ -52,20 +52,20 @@ const AdminCard = ({ createSnackbar, isSiteAdmin, involvementDescription, onAddM
   };
 
   const handleAddMember = async () => {
-    let memberEmail = username;
-    if (!memberEmail.toLowerCase().includes('@gordon.edu')) {
-      memberEmail = memberEmail + '@gordon.edu';
+    let formattedUsername = username.toLowerCase();
+    if (!formattedUsername.includes('@gordon.edu')) {
+      formattedUsername = formattedUsername.replace('@gordon.edu', '');
     }
```

Comment:
I think you inverted this process without inverting the `if`. Shouldn't this be 'if the username **includes** `'@gordon.edu'`'?

```javascript
    let formattedUsername = username.toLowerCase();
    if (formattedUsername.includes('@gordon.edu')) {
      formattedUsername = formattedUsername.replace('@gordon.edu', '');
    }
```

Also, I think this could be cleaned up in a couple of ways, now that I'm thinking about it:
1. Would it not be more efficient to always remove the ending `'@gordon.edu'`, rather then checking if it exists before removing it? Calling `replace()` with an unmatched pattern doesn't affect the output at all.
So you could rewrite it as:
```javascript
    const formattedUsername = username.toLowerCase().replace('@gordon.edu', '');
```
2. Technically, we want to replace `'@gordon.edu'` at the END of the string, not wherever we find it (although it should only exist at the end). We could do this with a RegExp pattern input to `replace`, but actually I think it would be clearer to use the `substring` and `lastIndexOf` functions on JS strings. The key thing to note is that `substring` returns the portion of the string from the `start` index (first param), up to but not including the `end` index. Also, I did some quick perf-testing and the latter is approximately 50-60% more performant than the former:
```javascript
    let domainIndex = username.toLowerCase().lastIndexOf('@gordon.edu');
    if (domainIndex === -1) domainIndex = username.length;
    const formattedUsername = username.substring(0, domainIndex);
```

Now at this point, this almost feels like it deserves to be a separate utility function:
```javascript
const stripDomain = (input, domain = '@gordon.edu') => {
    const domainIndex = input.toLowerCase().lastIndexOf(domain);
    
    // if domain was not found, return entire input
    if (domainIndex === -1) return input;
    
    // return input until domain
    return input.substring(0, domainIndex);
}
```

I think that at the very least we should adapt the 'always replace' style, but having this as it's own utility function might be really useful.

Reply:
Love it